test/unit_tests/sundials/test_sundials_datanode.cpp

```diff
@@ -0,0 +1,483 @@
+  err = SUNDataNode_Destroy(&child_node);
+  EXPECT_EQ(err, SUN_SUCCESS);
+}
+
+TEST_F(SUNDataNodeTest, RemoveChildWorksWhenEmpty)
+{
+  SUNErrCode err;
+  SUNDataNode root_node, child_node;
+  unsigned int num_elem = 5;
+
+  err = SUNDataNode_CreateList(SUNDATAIOMODE_INMEM, num_elem, sunctx, &root_node);
+  EXPECT_EQ(err, SUN_SUCCESS);
+
+  int integer_value = 5;
+  err = SUNDataNode_CreateLeaf(SUNDATAIOMODE_INMEM, mem_helper, sunctx,
+                               &child_node);
+  EXPECT_EQ(err, SUN_SUCCESS);
+
+  err = SUNDataNode_SetData(child_node, SUNMEMTYPE_HOST, SUNMEMTYPE_HOST,
+                            (void*)(&integer_value), sizeof(integer_value),
+                            sizeof(integer_value));
+  EXPECT_EQ(err, SUN_SUCCESS);
+
+  err = SUNDataNode_RemoveChild(root_node, 0, &child_node);
+  EXPECT_EQ(err, SUN_SUCCESS);
+
+  err = SUNDataNode_Destroy(&root_node);
+  EXPECT_EQ(err, SUN_SUCCESS);
+  err = SUNDataNode_Destroy(&child_node);
+  EXPECT_EQ(err, SUN_SUCCESS);
+}
+
+TEST_F(SUNDataNodeTest, GetChildWorks)
+{
+  SUNErrCode err;
+  SUNDataNode root_node, child_node;
+  unsigned int num_elem = 5;
+
+  err = SUNDataNode_CreateList(SUNDATAIOMODE_INMEM, num_elem, sunctx, &root_node);
+  EXPECT_EQ(err, SUN_SUCCESS);
+
+  int integer_value = 5;
+  err = SUNDataNode_CreateLeaf(SUNDATAIOMODE_INMEM, mem_helper, sunctx,
+                               &child_node);
+  EXPECT_EQ(err, SUN_SUCCESS);
+
+  err = SUNDataNode_SetData(child_node, SUNMEMTYPE_HOST, SUNMEMTYPE_HOST,
+                            (void*)(&integer_value), sizeof(integer_value),
+                            sizeof(integer_value));
+  EXPECT_EQ(err, SUN_SUCCESS);
+
+  err = SUNDataNode_AddChild(root_node, child_node);
+  EXPECT_EQ(err, SUN_SUCCESS);
+
+  err = SUNDataNode_GetChild(root_node, 0, &child_node);
+  EXPECT_EQ(err, SUN_SUCCESS);
+
+  EXPECT_EQ(integer_value, get_leaf_as_int(child_node));
+
+  err = SUNDataNode_Destroy(&root_node);
+  EXPECT_EQ(err, SUN_SUCCESS);
+  err = SUNDataNode_Destroy(&child_node);
+  EXPECT_EQ(err, SUN_SUCCESS);
+}
+
+TEST_F(SUNDataNodeTest, GetNamedChildWorks)
+{
+  SUNErrCode err;
+  SUNDataNode root_node, child_node;
+  unsigned int num_elem = 5;
+
+  err = SUNDataNode_CreateObject(SUNDATAIOMODE_INMEM, num_elem, sunctx,
+                                 &root_node);
+  EXPECT_EQ(err, SUN_SUCCESS);
+
+  int integer_value = 5;
+  err = SUNDataNode_CreateLeaf(SUNDATAIOMODE_INMEM, mem_helper, sunctx,
+                               &child_node);
+  EXPECT_EQ(err, SUN_SUCCESS);
+
+  err = SUNDataNode_SetData(child_node, SUNMEMTYPE_HOST, SUNMEMTYPE_HOST,
+                            (void*)(&integer_value), sizeof(integer_value),
+                            sizeof(integer_value));
+  EXPECT_EQ(err, SUN_SUCCESS);
+
+  err = SUNDataNode_AddNamedChild(root_node, "int_value", child_node);
+  EXPECT_EQ(err, SUN_SUCCESS);
+
+  err = SUNDataNode_GetNamedChild(root_node, "int_value", &child_node);
+  EXPECT_EQ(err, SUN_SUCCESS);
```

Comment:
Should this also test to ensure that the contents of `child_node` are correct, instead of just asking of the funciton returned an error message?